integration/kubernetes/init.sh

```diff
@@ -67,127 +100,208 @@ build_custom_stress_image()
+	fi
 
-sudo systemctl status "${cri_runtime}" --no-pager
+	sudo -E kubeadm init --config "${kubeadm_config_file}"
 
-echo "Init cluster using ${cri_runtime_socket}"
-kubeadm_config_template="${SCRIPT_PATH}/kubeadm/config.yaml"
-kubeadm_config_file="$(mktemp --tmpdir kubeadm_config.XXXXXX.yaml)"
+	mkdir -p "$HOME/.kube"
+	sudo cp "/etc/kubernetes/admin.conf" "$HOME/.kube/config"
+	sudo chown $(id -u):$(id -g) "$HOME/.kube/config"
+	export KUBECONFIG="$HOME/.kube/config"
 
-sed -e "s|CRI_RUNTIME_SOCKET|${cri_runtime_socket}|" "${kubeadm_config_template}" > "${kubeadm_config_file}"
-sed -i "s|KUBERNETES_VERSION|v${kubernetes_version/-*}|" "${kubeadm_config_file}"
-sed -i "s|CGROUP_DRIVER|${cgroup_driver}|" "${kubeadm_config_file}"
+	# enable debug log for kubelet
+	sudo sed -i 's/.$/ --v=4"/' /var/lib/kubelet/kubeadm-flags.env
+	echo "Kubelet options:"
+	sudo cat /var/lib/kubelet/kubeadm-flags.env
+	sudo systemctl daemon-reload && sudo systemctl restart kubelet
 
-trap 'sudo -E sh -c "rm -r "${kubeadm_config_file}""' EXIT
+	for i in $(seq 4)
+	do
+		kubectl get nodes && break || sleep $i
+	done
+	kubectl get pods
 
-if [ "${CI}" == true ] && [[ $(wc -l /proc/swaps | awk '{print $1}') -gt 1 ]]; then
-	grep -q zram /proc/swaps && echo "# zram swap disabled"  | sudo tee /etc/systemd/zram-generator.conf
-	sudo swapoff -a || true
-fi
+}
 
-#reinstall kubelet to do deep cleanup
-if [ "${BAREMETAL}" == true -a "$(command -v kubelet)" != "" ]; then
-	info "reinstall kubeadm, kubelet before initialize k8s"
-	bash -f "${SCRIPT_PATH}/../../.ci/install_kubernetes.sh"
-fi
+# Start the CRI runtime service.
+#
+# Arguments:
+#	$1 - the CRI service name (mandatory).
+#	$2 - the expected service socket path (mandatory).
+#
+start_cri_runtime_service() {
+	local cri=$1
+	local socket_path=$2
+	# Number of check tentatives.
+	local max_cri_socket_check=5
+	# Sleep time between checks.
+	local wait_time_cri_socket_check=5
+
+	# stop containerd first and then restart it
+	if systemctl is-active --quiet containerd; then
+		info "Stop containerd service"
+		sudo systemctl stop containerd
+	fi
 
-sudo -E kubeadm init --config "${kubeadm_config_file}"
+	sudo systemctl enable --now ${cri}
 
-mkdir -p "$HOME/.kube"
-sudo cp "/etc/kubernetes/admin.conf" "$HOME/.kube/config"
-sudo chown $(id -u):$(id -g) "$HOME/.kube/config"
-export KUBECONFIG="$HOME/.kube/config"
+	for i in $(seq ${max_cri_socket_check}); do
+		#when the test runs two times in the CI, the second time crio takes some time to be ready
+		sleep "${wait_time_cri_socket_check}"
+		[ -e "${socket_path}" ] && break
+		echo "Waiting for cri socket ${socket_path} (try ${i})"
+	done
 
-# enable debug log for kubelet
-sudo sed -i 's/.$/ --v=4"/' /var/lib/kubelet/kubeadm-flags.env
-echo "Kubelet options:"
-sudo cat /var/lib/kubelet/kubeadm-flags.env
-sudo systemctl daemon-reload && sudo systemctl restart kubelet
+	sudo systemctl status "${cri}" --no-pager || \
+		die "Unable to start the ${cri} service"
```

Comment:
Would have been enough bco `errexit` IMO, but maybe the error message makes it a little nicer.